src/NzbDrone.Core/Parser/LanguageParser.cs

```diff
@@ -25,7 +26,11 @@ public static class LanguageParser
         private static readonly Regex CaseSensitiveLanguageRegex = new Regex(@"(?:(?i)(?<!SUB[\W|_|^]))(?:(?<lithuanian>\bLT\b)|(?<czech>\bCZ\b)|(?<polish>\bPL\b)|(?<bulgarian>\bBG\b)|(?<slovak>\bSK\b))(?:(?i)(?![\W|_|^]SUB))",
                                                                 RegexOptions.Compiled);
 
-        private static readonly Regex SubtitleLanguageRegex = new Regex(".+?[-_. ](?<iso_code>[a-z]{2,3})([-_. ](?<tags>full|forced|foreign|default|cc|psdh|sdh))*$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        private static readonly Regex SubtitleLanguageRegex = new Regex(".+?([-_. ](?<tags>full|forced|foreign|default|cc|psdh|sdh))*[-_. ](?<iso_code>[a-z]{2,3})([-_. ](?<tags>full|forced|foreign|default|cc|psdh|sdh))*$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+        private static readonly Regex SubtitleLanguageTitleRegex = new Regex(".+?(\\.((?<tags1>full|forced|foreign|default|cc|psdh|sdh)|(?<iso_code>[a-z]{2,3})))*\\.(?<title>[^.]*)(\\.((?<tags2>full|forced|foreign|default|cc|psdh|sdh)|(?<iso_code>[a-z]{2,3})))*$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```

Comment:
I don't think we can assume title will never have a period

Reply:
Well the reason that is necessary is to check if the title is first. If it isn't first, parsing is a little easier because the other tags are sufficiently specific as that they should mostly discard everything that comes at the end of an episode title and isn't a tag/subtitle title. The question of whether there is a dot only comes into play if the title is first(or non-existent). To guard against incorrectly parsing a title when there is none, the heuristic that if the episode file path contains the string that was parsed as a subtitle file title, it will not be used. See some of the testcases for examples of that.

Reply:
What about something like `series.title.s01e01.my.name.is.default.eng.forced.srt`? If the episode file is `My Name is Default` this will still parse it incorrectly while trying to guard against it and it should still be cleaned up via an augmenter.

Reply:
This is why I'm actually checking both the original and current file name for matches. I guess I could look at the entire history of renames of the file(somehow, need to see how that might be possible)? Or at least at the previous name.